master/internal/configpolicy/task_config_policy.go

```diff
@@ -83,7 +84,7 @@ func CheckExperimentConstraints(
 
 	if constraints.ResourceConstraints != nil && constraints.ResourceConstraints.MaxSlots != nil {
 		// users cannot specify number of slots for an experiment
-		slotsRequest := 0
+		slotsRequest := *constraints.ResourceConstraints.MaxSlots
```

Comment:
This is an unrelated change. I was thinking about it and this seems more resilient; an admin is allowed to set a constraint of `max_slots: -1`; we shouldn't start rejecting experiments in that case.

I can put this in a separate PR if preferred.